### PR TITLE
Add more reference tests

### PR DESCRIPTION
**Pull request by fingolfin**

### CODE REVIEW

**Comment by fingolfin** [comment]
The test failure is caused by an extra whitespace in the output, which I missed because I was testing with a modified version of `BibInternal` which fixes that spacing issue, among other things.

Which raises some questions how to proceed (and IMHO highlights one the drawbacks of splitting everything into so many different packages): I could submit the fix for BibIntern first, then wait till that is merged and released, then update this PR. But then ideally there should also be an explicit dependency on the relevant BibIntern version?

Or I could modify this PR to match the current BibIntern, and it gets merged first. But then once BibIntern receives that whitespace fix, and gets released, it'll silently break the test suite of this package. Of course it'll be easy to fix that, and it's fine as long as everything happens around the same time; but it can be more of a nuisance if more time passes (to me this suggests that BibIntern should *also* run the test suite of Bibliography.jl, to see if a change there causes regressions there... i.e. a kind of integration test. Of course one can then wonder whether to test against the latest Bibliography release; or its master branch; or both; or something else).

Anyway: how would you like to proceed?

**Comment by fingolfin** [comment]
I guess the safest way to proceed would be as follows: merge PR #28, then release Bibliography.jl; then merge <https://github.com/Humans-of-Julia/BibInternal.jl/pull/12> (and perhaps <https://github.com/Humans-of-Julia/BibInternal.jl/pull/13>) and release `BibInternal.jl`; and only then merge this PR here, and perhaps also require `BibInternal.jl` version >= WHATEVER-IT-IS

**Comment by Azzaare** [comment]
The original reason to split the packages into 3 was to have BibParser separated from the rest. And that was because I was parsing BibTeX thanks to Automa.jl (which was really nice and clean). However, the precompile time of the automa was crazy (5/10 minutes).
Now that I wrote a better parser manually, I kind of regret the split now, but at least if someone add a very slow to precompile parser to BibParser, it is not too bad.

Back on track, I will increase BibInternal to v0.3.0, so that it won't break Bibliography until I update the compat file. (but first I will test everything ^^)

Also, you're right about testing everything in BibInternal too. I am actually planning to append all the tests in BibInternal, BibParser, Bibliography, DocumenterCitations, and StaticWebpages together as they are the main packages currently related to Bibliography.

**Comment by Azzaare** [comment]
I will be offline for a while, but the new version of BibInternal should come soon.

In any case, if you want to update the Project.toml, it should look like this:
```toml
name = "Bibliography"
uuid = "f1be7e48-bf82-45af-a471-ae754a193061"
authors = ["azzaare <jf@baffier.fr>"]
version = "0.2.12"

[deps]
BibInternal = "2027ae74-3657-4b95-ae00-e2f7d55c3e64"
BibParser = "13533e5b-e1c2-4e57-8cef-cac5e52f6474"
DataStructures = "864edb3b-99cc-5e75-8d2d-829cb0a9cfe8"

[compat]
BibInternal = "0.3"
BibParser = "0.1"
DataStructures = "0.18"
julia = "1.4, 1.5, 1.6"
```
Only the `version` and `[compat]->BibInternal` lines should be updated!

File: test/test.bib [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+ howpublished  = {Distributed at the Stockholm Tourist Office},
+ publisher     = {Penguin Random House},
+ address       = {New York, NY},
+ author        = {Swetla, Maria},
+ year          = {2015},
+ month         = {{July}},
+ title         = {Canoe tours in {S}weden}
+}
+
+@misc{,
```
Comment (fingolfin) [inline test/test.bib:39]: Note the missing key here; this is issue #29 again

**Comment by Azzaare** [comment]
I have updated the ref tests according to the last version of BibParser (that fixes the missing keys)

@fingolfin, merging https://github.com/fingolfin/Bibliography.jl/pull/1 should fix this PR :)

**Comment by fingolfin** [comment]
Done

**Comment by Azzaare** [comment]
We have an error from a bad compat file in doc, which is fine (I will fix it later).

We also have one interesting error for x_86 architecture due to a different ordering of the hashtable on 32 bits systems.

I will merge that, but I will have to print the field in alphabetical order eventually ;)

**Pull request merged**